src/main/java/seedu/address/model/event/Event.java

```diff
@@ -0,0 +1,92 @@
+package seedu.address.model.event;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an Event in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Event {
+
+    // Identity fields
+    private final EventName eventName;
+
+    // Data fields
+    private final Location location;
+    private final EventDate eventDate;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Event(EventName eventName, EventDate eventDate, Location location) {
+        requireAllNonNull(eventName, eventDate, location);
+        this.eventName = eventName;
+        this.eventDate = eventDate;
+        this.location = location;
+    }
+
+    public EventName getName() {
+        return eventName;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+    public EventDate getDate() {
+        return eventDate;
+    }
+
+    /**
+     * Returns true if both events have the same name.
+     * This defines a weaker notion of equality between two events.
+     */
+    public boolean isSameEvent(Event otherEvent) {
+        if (otherEvent == this) {
+            return true;
+        }
+
+        return otherEvent != null
+                && otherEvent.getName().equals(getName());
+    }
```

Comment:
Should the constraint here be stricter, as in if the Event has same name & same date & same location then is same event. 
Or is the same name only sufficient already

Reply:
Hmm what do the other think about this?

Reply:
I think that enforcing same date and location would be important and should not be left out. This will be crucial as certain events can have the same name but different dates, especially for multi-day events. For example, Alice can be involved in a camp for Day 1 only while Bob can be involved in the same camp for Day 2 only, etc.